File: freighter/ts/src/lib/util/log.ts
// Copyright 2022 Synnax Labs, Inc.
//
// Use of this software is governed by the Business Source License included in the file
// licenses/BSL.txt.
//
// As of the Change Date specified in that file, in accordance with the Business Source
// License, use of this software will be governed by the Apache License, Version 2.0,
// included in the file licenses/APL.txt.

import { Middleware } from "../middleware";

export const logMiddleware = (): Middleware => {
  return async (md, next) => {
    console.log(JSON.stringify(md, undefined, 2));
    const err = await next(md);
    if (err != null) {
      console.log(err);
    }
    return err;
  };
};

File: synnax/pkg/service/workspace/log/log.go
// Copyright 2025 Synnax Labs, Inc.
//
// Use of this software is governed by the Business Source License included in the file
// licenses/BSL.txt.
//
// As of the Change Date specified in that file, in accordance with the Business Source
// License, use of this software will be governed by the Apache License, Version 2.0,
// included in the file licenses/APL.txt.

package log

import (
	"github.com/google/uuid"
	"github.com/synnaxlabs/x/gorp"
)

// Log is the data for a visualization component used to view a log of telemetry over
// time.
type Log struct {
	// Key is a unique identifier for the log.
	Key uuid.UUID `json:"key" msgpack:"key"`
	// Name is a human-readable name for the log.
	Name string `json:"name" msgpack:"name"`
	// Data is JSON-encoded data for the log.
	Data string `json:"data" msgpack:"data"`
}

var _ gorp.Entry[uuid.UUID] = Log{}

// GorpKey implements gorp.Entry.
func (l Log) GorpKey() uuid.UUID { return l.Key }

// SetOptions implements gorp.Entry.
func (l Log) SetOptions() []any { return nil }
